### ISSUE TITLE
Code Quality/Optimizations

### ISSUE THREAD

**Issue by rgroves** [tomassirio/ListBot#153]
There were some code quality/optimization issues discovered by @RazCrimson (as noted in #150 in [this comment](https://github.com/tomassirio/ListBot/issues/150#issuecomment-716821751)). Copying the relevant info here as well:

"_As for the redundant initializations, it was Webstorm reporting to me while I made commits. We should have like half a dozen or so redundant initializations and the minutes variable being assigned 0 in line number 72 of poll.js had no effect at all.

I will check the code base later and open another issue if there are other stuff like this._"

We should try to identify the code analysis rules that were being used and incorporate those into the current linting rules in place for this codebase.

**Comment by rgroves** [comment]
@RazCrimson are you able to help identify what rules Webstorm was using when it reported the code quality (redundant initialization and/or other) issues? I'd like to try to identify them and see if we can turn on equivalent rules in the eslint tool that is in use for the ListBot codebase.

**Comment by RazCrimson** [comment]
@tomassirio @rgroves Can anyone of you review and merge master with #149 ? Waiting for it to be done so that I wont have to merge my changes, in case there are any conflicts.

As for the ESlint rules, I can't find a equivalent in the documentation for what Webstorm is detecting here.

**Comment by rgroves** [comment]
@RazCrimson I'm waiting for two minor changes to be worked into #149 before it can be merged.

If you'd like to continue working, one way to do it is as follows: 

You can add a remote for anontyro's repo:
```
#  add anontyro's repo as a remote with the name "workaround"
git remote add workaround git@github.com:anontyro/ListBot.git

#  fetch from workaround
git fetch workaround

# Checkout and create a tracking branch for the branch containing anontryo's work that you need to continue off of.
git checkout -t workaround/feature/add-colour-to-list
```

Once that is done, you'll be on the feature/add-colour-to-list branch and can continue your work on that branch. Be aware there will be two adjustments you'll want to make in `src/utils/utils.js`. Those are outlined in [my last review comments on 149](https://github.com/tomassirio/ListBot/pull/149#pullrequestreview-518250627). Then commit those changes, (locally only don't push any of this stuff up to your fork), and then ignore this commit later when cherry picking (see below).

When #149 is finally merged in you can sync your local working copy's master. Once that is done create a new branch from master for this issue's work and cherry pick the commits of the work you did in feature/add-colour-to-list. Once you do this you'll no longer need the stuff you did for a workaround.

Finally you can get rid of all of anontyro's stuff from your repo with:
```
git checkout feature/add-colour-to-list
git branch --unset-upstream
git checkout master
git remote remove workaround
git branch -D feature/add-colour-to-list
```

**Comment by rgroves** [comment]
@RazCrimson Also, how familiar are you with Webstorm? I'm trying to look into what it is using for doing it's checks. I found something that said it integrates with ESLint, but if as you say there is no equivalent in ESLint, it must be using something else. Would love to see a list of its rules. Maybe it's using additional plug-ins?

If you had time to dig into that and report back, it would be very much appreciated. If not, if it will allow you to copy and paste the rule violations it's reporting with ListBot's codebase, please provide them in a comment or attachment here. That would be great as well. Thanks!

**Comment by RazCrimson** [comment]
@rgroves 
I have 6 warnings for redundant initialization like this:
![image](https://user-images.githubusercontent.com/52282402/97434253-bf20cb00-1944-11eb-9479-c7d47e75b3c5.png)

Do u see the resultMessage being assigned values in all if..else blocks? So, that makes assigning a initial value to it redundant.

Then do you remember that awaits that I added some of my initial commits. They are caught by WebStorm too.
![image](https://user-images.githubusercontent.com/52282402/97435728-d95ba880-1946-11eb-8c5e-887edda02344.png)

Actually adding those awaits, wont really impact anything for our usecase, it will just make code follow general standards.

Also in `src/utils/index.js` in line number `18`, there is a typo for `mongodb`, code has `monogdb`. I cant find any usages for it right now, will change that too.

Then as for how we can add these features to ESLint rules, I dont think we can. Webstorm can integrate with ESLint. But the warnings given here are found by a 'Code Inspection' feature which is natively available in WebStorm. It works based on templates. Refer this: https://www.jetbrains.com/help/webstorm/code-inspection.html

**Comment by rgroves** [comment]
Okay, I'm on the fence about warnings (especially the ones labeled 'weak warnings') for rules we can't enforce. Just because if we're not able to enforce them they will likely creep into the codebase over time, then only a WebStorm user will be able to identify them. 

With that said, if you want to make the changes and they improve the codebase, I'm not entirely against it either.

I haven't heard back on the PR with the outstanding changes we're waiting on so I think I will move forward my merging it and making the two changes as a fixup to that PR. Then if you'd like you can continue to work this issue.

**Comment by rgroves** [comment]
I just wanted to add a note here for future reference. I was struggling with some of the `await` keyword additions, so looked into the reasons behind why WebStorm's code inspection warns when they are missing. I found this:
[JetBrains reasoning for "Code Inspection: Missing await for an async function call"](https://www.jetbrains.com/help/phpstorm/javascript-and-typescript-missing-await-for-an-async-function-call.html)

I've recapped the reasoning below.

> This is generally not necessary, but it gives two main benefits:
> 1. You won't forget to add 'await' when surrounding your code with try-catch.
> 2. Having explicit 'await' helps V8 runtime to provide async stack traces
